spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -799,41 +820,33 @@ def _calc_run_time(self, run_time):
         """
         if run_time is None:
             return None, None
-        machine_time_step_ms = (
-            self.machine_time_step / MICRO_TO_MILLISECOND_CONVERSION)
-        n_machine_time_steps = math.ceil(run_time / machine_time_step_ms)
-        calc_run_time = n_machine_time_steps * machine_time_step_ms
-
-        # Allow for minor float errors
-        if abs(run_time - calc_run_time) > 0.00001:
-            logger.warning(
-                "Your requested runtime of {}ms "
-                "is not a multiple of the machine time step of {}ms "
-                "and has therefor been rounded up to {}ms",
-                run_time, machine_time_step_ms, calc_run_time)
-
+        n_machine_time_steps = self.__timesteps(run_time)
         total_run_timesteps = (
             self._current_run_timesteps + n_machine_time_steps)
+        machine_time_step_ms = (
+            self.machine_time_step / MICRO_TO_MILLISECOND_CONVERSION)
         total_run_time = (
             total_run_timesteps * machine_time_step_ms *
             self.time_scale_factor)
 
-        # Convert dt into microseconds and divide by
-        # realtime proportion to get hardware timestep
+        # Convert dt into microseconds and multiply by
+        # scale factor to get hardware timestep
         hardware_timestep_us = int(round(
-            float(self.machine_time_step) / float(self.time_scale_factor)))
+            float(self.machine_time_step) * float(self.time_scale_factor)))
```

Comment:
Are you sure this is now the right way round?  I don't remember this being wrong previously...

Reply:
I forgot I had changed this, but the timestep on the machine is definitely multiplied by the timescale factor, as otherwise it would get smaller...

Reply:
Ah, okay.  I forget every so often that the time_scale_factor isn't a fraction but an integer... so happy to merge this now.